src/low-level/imap/mailimap_extension.c

```diff
@@ -166,7 +166,8 @@ mailimap_extension_data_free(struct
   if (data == NULL)
     return;
 
-  if (data->ext_extension != NULL)
+  if (data->ext_extension != NULL && data->ext_data != NULL)
+    /* ext_free() includes free(data) */
     data->ext_extension->ext_free(data);
   else
```

Comment:
Could you remove the else here?
I think free(data) always need to be executed.

Reply:
Tested it. This causes double free because ext_free also free(data)
in file low-level/imap/acl-types.c `mailimap_acl_free`

Reply:
Ah! very true!